exercises/ruby-1-b/ruby_1_b_test.rb

```diff
@@ -1,34 +1,78 @@
 
-# Common test data version: 1.2.0 4fc1acb
-class SentenceTest < Minitest::Test
-  def test_one_word_with_one_vowel
-    assert_equal "a", word_with_most_vowels("a")
+class LibraryTest < Minitest::Test
+  def setup
+    @library = Library.new
   end
 
-  def test_one_word_with_one_vowel_and_one_consonant
-    assert_equal "at", word_with_most_vowels("at")
+  def test_lookup_title_returns_correct_title
+    assert_equal "The Adventures of Tom Sawyer", @library.lookup_title("9780191604928")
   end
 
-  def test_one_word_with_two_vowels_and_one_consonant
-    assert_equal "tea", word_with_most_vowels("tea")
+  def test_lookup_author_returns_correct_author
+    assert_equal "Plato", @library.lookup_author("9780718198916")
   end
 
-  def test_longest_word_is_not_word_with_most_vowels
-    assert_equal "area", word_with_most_vowels("the area of a circle")
+  def test_lookup_publication_year_returns_correct_publication_year
+    assert_equal "2004", @library.lookup_publication_year("9780974514055")
   end
 
-  def test_last_word_is_word_with_most_vowels
-    assert_equal "vegetables", word_with_most_vowels("extremely fresh vegetables")
+  def test_add_stock_changes_correct_stock
+    @library.add_stock!("9781986431484", 5)
+    assert_equal 5, @library.lookup_stock("9781986431484")
   end
 
-  def test_mixed_casing
-    assert_equal "YOUR", word_with_most_vowels("TEST YOUR MIGHT")
+  def test_add_stock_does_not_change_stock_for_other_book
+    @library.add_stock!("9781986431484", 5)
+    assert_equal 0, @library.lookup_stock("9780446310789")
   end
 
-  def test_ignore_word_without_vowels
-    assert_equal "great", word_with_most_vowels("great rhythm")
+  def test_book_in_stock_when_no_stock_has_been_added
+    refute @library.book_in_stock?("9781986431484")
+  end
+
+  def test_book_in_stock_when_num_in_stock_is_at_least_one
+    @library.add_stock!("9781986431484", 1)
+    assert @library.book_in_stock?("9781986431484")
+  end
+
+  def test_borrowing_book_with_stock_of_one
+    @library.add_stock!("9780330319089", 1)
+    @library.borrow!("9780330319089")
+    refute @library.book_in_stock?("9780330319089")
+  end
+
+  def test_borrowing_book_with_stock_more_than_one
+    @library.add_stock!("9780330319089", 2)
+    @library.borrow!("9780330319089")
+    assert @library.book_in_stock?("9780330319089")
   end
-end
 
+  def test_put_back_when_book_has_not_been_borrowed
+    @library.put_back!("9780330319089")
+    assert_equal 0, @library.lookup_stock("9780330319089")
+  end
+
+  def test_put_back_when_book_has_been_borrowed
+    @library.add_stock!("9780330319089", 1)
+    @library.borrow!("9780330319089")
+    @library.put_back!("9780330319089")
+    assert_equal 1, @library.lookup_stock("9780330319089")
+  end
+
+  def test_books_in_stock
+    isbns = ["9780544003415", "9780140283334", "9780451524935"]
+    isbns.each { |isbn| @library.add_stock!(isbn, 1) }
+    assert_includes @library.books_in_stock, "9780544003415"
```

Comment:
This could probably be `assert_equal ibsns.sort, @library.books_in_stock.sort` and then delete the next few lines.

Reply:
Good idea 👍 I'll update it